tilequeue/format/__init__.py

```diff
@@ -38,27 +40,21 @@ def format_tile(self, tile_data_file, feature_layers, coord, bounds_merc,
 
 # consistent facade around all tilestache formatters that we use
 def format_json(fp, feature_layers, coord, bounds_merc, bounds_wgs84):
-    # TODO a lot of serializing/deserializing can be reduced here
-    # this is a faithful port for how it's done in tilestache now
     if len(feature_layers) == 1:
-        json_encode(fp, feature_layers[0]['features'], coord.zoom)
+        json_encode_single_layer(fp, feature_layers[0]['features'], coord.zoom)
         return
-    names = []
-    layers = []
-    for feature_layer in feature_layers:
-        names.append(feature_layer['name'])
-        out = StringIO()
-        json_encode(out, feature_layer['features'], coord.zoom)
-        # out now contains a json serialized result
-        # now we deserialize it, so that it can be combined with the
-        # merge function
-        deserialized_features = loads(out.getvalue())
-        layers.append(deserialized_features)
-    json_merge(fp, names, layers, None, coord)
+    else:
+        layer_features = []
+        for feature_layer in feature_layers:
+            layer_name = feature_layer['name']
+            features = feature_layer['features']
+            layer_features.append((layer_name, features))
```

Comment:
Just that it was only used for iterating and not lookups, so a list of tuples was more lightweight. But you're right that by taking a dict we ensure that it'll be a unique list of keys that get passed in, so we wouldn't have the situation where duplicates overwrote earlier layers in this case. Updated in eff37b1.